packages/offix-client/src/OfflineClient.ts

```diff
@@ -100,21 +101,28 @@ export class OfflineClient implements ListenerProvider {
    *
    * @param options the MutationHelperOptions to create the mutation
    */
-  public offlineMutation<T = any, TVariables = OperationVariables>(
+  public async offlineMutate<T = any, TVariables = OperationVariables>(
     options: MutationHelperOptions<T, TVariables>): Promise<FetchResult<T>> {
     if (!this.apolloClient) {
       throw new Error("Apollo offline client not initialised before mutation called.");
+    }
+
+    const mutationOptions = createMutationOptions<T, TVariables>(options);
+    const isOffline = await this.config.networkStatus.isOffline();
+    if (isOffline) {
+      // FIXME use internal part of the offline engine for this
+      OfflineMutationsHandler.markAsOffline(mutationOptions);
```

Comment:
In this line you pass the whole operation vars but only operate on the context in OfflineMutationsHandler, I think this line should be mutationOptions.context